data_steward/validation/ehr_merge.py

```diff
@@ -0,0 +1,200 @@
+    :dataset_id: dataset with the tables
+    :returns: string if visit table failed; otherwise none
+
+    """
+    # # list of hpos with visit table and creating visit id mapping table queries
+    visit_hpo_queries = []
+    for hpo in hpos_with_visit:
+        visit_hpo_queries.append(VISIT_ID_HPO_BLOCK % locals())
+    union_all_blocks = '\n UNION ALL'.join(visit_hpo_queries)
+    visit_mapping_query = VIST_ID_MAPPING_QUERY_SKELETON % locals()
+    logging.info('Loading ' + VISIT_ID_MAPPING_TABLE)
+    query_result = query(visit_mapping_query,
+                         destination_table_id=VISIT_ID_MAPPING_TABLE,
+                         write_disposition='WRITE_TRUNCATE')
+    visit_mapping_query_job_id = query_result['jobReference']['jobId']
+    incomplete_jobs = bq_utils.wait_on_jobs([visit_mapping_query_job_id], retry_count=10)
+    if len(incomplete_jobs) == 0:
+        query_result = bq_utils.get_job_details(visit_mapping_query_job_id)
+        if 'errors' in query_result['status']:
+            logging.error('{} load failed!'.format(VISIT_ID_MAPPING_TABLE))
+            return "visit mapping table failed"
+    else:
+        logging.error('{} load taking too long!'.format(VISIT_ID_MAPPING_TABLE))
+        raise RuntimeError("visit mapping table taking too long. see job id: {}".format(visit_mapping_query_job_id))
+
+
+def merge(dataset_id, project_id):
+    """merge hpo ehr data
+
+    :dataset_id: source and target dataset
+    :project_id: project in which everything happens
+    :returns: string with job details
+
+    """
+    existing_tables = bq_utils.list_dataset_contents(dataset_id)
+    hpos_to_merge = []
+    hpos_with_visit = []
+    for item in resources.hpo_csv():
+        hpo_id = item['hpo_id']
+        if hpo_id + '_person' in existing_tables:
+            hpos_to_merge.append(hpo_id)
+        if hpo_id + '_visit_occurrence' in existing_tables:
+            hpos_with_visit.append(hpo_id)
+
+    create_mapping_table(hpos_with_visit, project_id, dataset_id)
+
+    jobs_to_wait_on = []
+    for table_name in common.CDM_TABLES:
+        q = construct_query(table_name, hpos_to_merge, hpos_with_visit, project_id, dataset_id)
+        logging.info('Merging table: ' + table_name)
+        query_result = query(q, destination_table_id='unioned_ehr_'+table_name, write_disposition='WRITE_TRUNCATE')
+        query_job_id = query_result['jobReference']['jobId']
+        jobs_to_wait_on.append(query_job_id)
+
+    incomplete_jobs = bq_utils.wait_on_jobs(jobs_to_wait_on, retry_count=10)
+    if len(incomplete_jobs) == 0:
+        status_list = [bq_utils.get_job_details(job_id)['status'] for job_id in jobs_to_wait_on]
+        table_errors = [list(common.CDM_TABLES)[ind] for ind, _ in enumerate(jobs_to_wait_on)
+                        if 'errors' in status_list[ind]]
+        required_table_error_list = [table_name for table_name in table_errors
+                                     if table_name in common.REQUIRED_TABLES]
```

Comment:
@aahladmanas @cukarthik [common.py](https://github.com/aahladmanas/curation/blob/feature-branch/ehr-merge/data_steward/common.py#L22) only lists `person` as a required table. I don't think we want to consider the process successful with just the union of the `person` tables. This is especially the case if there were other tables our process failed to include due to errors not under a particular HPO's control (e.g. a downtime, a BigQuery error, a bug in our code, a problem with another HPO's `visit_occurrence` table, etc.). Further, if we are making the assumption that the source CDM tables that get to this point are worthy of including in the CDR, shouldn't loss of any of these tables be considered an error in our system?